bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsDeviceHandler.java

```diff
@@ -48,13 +49,16 @@ public void initialize() {
         ElroConnectsDeviceConfiguration config = getConfigAs(ElroConnectsDeviceConfiguration.class);
         deviceId = config.deviceId;
 
+        Bridge bridge = getBridge();
         ElroConnectsBridgeHandler bridgeHandler = getBridgeHandler();
 
-        if (bridgeHandler != null) {
+        if ((bridge != null) && (bridgeHandler != null) && (bridge.getStatus() == ThingStatus.ONLINE)) {
             bridgeHandler.setDeviceHandler(deviceId, this);
             updateProperties(bridgeHandler);
             updateDeviceName(bridgeHandler);
             refreshChannels(bridgeHandler);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
         }
```

Comment:
You are already setting the thing status to OFFLINE in getBridgeHandler() if there is no bridge or bridge handler is null.
Rather use BRIDGE_OFFLINE is the last updateStatus.
So I would rather suggest the following code:
```
        ElroConnectsBridgeHandler bridgeHandler = getBridgeHandler();
        if (bridgeHandler == null) {
             // Thing status has already been updated in getBridgeHandler()
             return;
        }

        if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
            bridgeHandler.setDeviceHandler(deviceId, this);
            updateProperties(bridgeHandler);
            updateDeviceName(bridgeHandler);
            refreshChannels(bridgeHandler);
        } else {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
        }
```

I do not see where the status is set to ONLINE. Maybe it should be added there.

I also found that two methods `updateStatus` were override just to call the super method. I do not understand why.

Reply:
@lolodomo Thank you for the simplification, I changed it.

The thing status is utlimately set in the refreshChannels call.
I need to override the methods because they are protected in the BaseThingHandler. I call these from outside the package.

Reply:
Shouldn't you cover the case your device id is invalid ?
Something like that:
```
        if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
            ElroConnectsDevice device = bridgeHandler.getDevice(deviceId);
            if (device != null) {
                bridgeHandler.setDeviceHandler(deviceId, this);
                updateProperties(bridgeHandler);
                updateDeviceName(bridgeHandler);
                refreshChannels(bridgeHandler);
            } else {
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "xxxxxx");
            }
        } else {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
        }
```

Reply:
@lolodomo Indeed, good catch. I typically use discovery in testing, and then the problem does not appear. And as the refresh did not work in the case of a wrong deviceId, the thing stayed in initializing state.
I actually needed one more check. The user could create a thing of the wrong type, even if the deviceId is valid in the controller. So I am now checking for that as well.
For safety, I have inserted setting thing status to unknown before calling refresh. If something is still not covered, it would not keep showing INITIALIZING.
Many thanks for your eye for detail and your torough review.